File: algorithm/boj/_7000/7568.c
#include <stdio.h>

int main() {
    int n, i, j, grade;
    scanf("%d", &n);
    int weight[n];
    int height[n];
    for(i=0; i<n; i++) {
        scanf("%d %d", &weight[i], &height[i]);
    }
    for(i=0; i<n; i++) {
        grade=1;
        for(j=0; j<n; j++) {
            if((i!=j)&(weight[i]<weight[j]&height[i]<height[j])) {
                grade++;
            }
        }
        printf("%d ", grade);
    }
    printf("\n");
    return 0;
}

File: algorithm/boj/_7000/7568.py
import sys

if __name__ == "__main__":
    N = int(sys.stdin.readline())
    weight = []
    height = []
    
    for i in range(N):
        x, y = map(int, sys.stdin.readline().split(" "))
        weight.append(x)
        height.append(y)
    
    for i in range(N):
        rank = 1
        for j in range(N):
            if weight[i] < weight[j] and height[i] < height[j]: rank += 1
        print(rank, end=" ")